app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/map/offline/ManageOfflineMapActivityTest.kt

```diff
@@ -109,80 +67,30 @@ class ManageOfflineMapActivityTest {
         Assert.assertEquals("ch.epfl.sdp.drone3d", appContext.packageName)
     }
 
-    /**
-     * Make sure the tile count is initialized to 0
-     */
     @Test
-    fun tileCountIsInitializedTo0() {
-        SystemClock.sleep(2000) //Because the view is a observer and we need some time to make it get the value
-        onView(withId(R.id.tiles_used))
-            .check(matches(withText("0/6000")))
-    }
-
-    /**
-     * This test tests most of the UI functionality at once. The reason is that the downloading takes
-     * quite a long time and that it is done asynchronously, thus it makes the testing faster if we don't
-     * have to wait for most of the tests to download some tiles. Furthermore, mocking the OfflineMapSaverImpl
-     * is not possible either, because Mapbox's OfflineRegion cannot be instantiated since its constructor
-     * is private.
-     */
-    @Test
-    fun userBehaviorTest() {
-        SystemClock.sleep(2000) //Need to wait for the map to be downloaded. Doesn't work with later callback only
-
-        //Zoom to make the download happens faster so that we don't have to wait to sleep for too long
-        var counter = CountDownLatch(1)
-
-        activityRule.scenario.onActivity { activity ->
-            activity.mapView.getMapAsync { mapboxMap ->
-                mapboxMap.setStyle(Style.MAPBOX_STREETS) { _ ->
-                    MapboxUtility.zoomOnCoordinate(ZOOM_LOCATION, mapboxMap, ZOOM_VALUE)
-                    counter.countDown()
-
-                }
-            }
-        }
-
-        assert(counter.await(TIMEOUT, TimeUnit.SECONDS))
+    fun cannotEnterEmptyStringForRegionName() {
+        SystemClock.sleep(500)
```

Comment:
Is it not possible to wait until something is displayed ? This kind of random waits make tests non deterministic.